youtube_dl/extractor/ximalaya.py

```diff
@@ -0,0 +1,184 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+import itertools
+
+from ..compat import (
+    compat_str,
+)
+
+from .common import InfoExtractor
+
+
+class XimalayaBaseIE(InfoExtractor):
+    _GEO_COUNTRIES = ['CN']
+
+
+class XimalayaIE(XimalayaBaseIE):
+    IE_NAME = 'ximalaya'
+    IE_DESC = 'ximalaya.com'
+    _VALID_URL = r'https?://(?:www\.|m\.)?ximalaya\.com/(?P<uid>[0-9]+)/sound/(?P<id>[0-9]+)/?'
+    _USER_URL_FORMAT = 'http://www.ximalaya.com/zhubo/%i/'
+    _TESTS = [
+        {
+            'url': 'http://www.ximalaya.com/61425525/sound/47740352/',
+            # 'md5': 'TODO: md5 sum of the first 10241 bytes of the video file (use --test)',
+            'info_dict': {
+                'id': '47740352',
+                'ext': 'm4a',
+                'title': '261.唐诗三百首.卷八.送孟浩然之广陵.李白',
+                'description': 'contains:孤帆远影碧空尽，惟见长江天际流。',
+                'uploader': '小彬彬爱听书',
+                'uploader_id': 61425525,
+                'view_count': int,
+                'like_count': int,
+            }
+        },
+        {
+            'url': 'http://m.ximalaya.com/61425525/sound/47740352/',
+            'info_dict': {
+                'id': '47740352',
+                'ext': 'm4a',
+                'title': '261.唐诗三百首.卷八.送孟浩然之广陵.李白',
+                'description': 'contains:孤帆远影碧空尽，惟见长江天际流。',
+                'uploader': '小彬彬爱听书',
+                'uploader_id': 61425525,
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+
+        is_m = 'm.ximalaya' in url
+
+        audio_id = self._match_id(url)
+        webpage = self._download_webpage(url, audio_id,
+                                         note='Download sound page for %s' % audio_id,
+                                         errnote='Unable to get sound page')
+
+        audio_info_file = 'http://m.ximalaya.com/tracks/%s.json' % audio_id
+        audio_info = self._download_json(audio_info_file, audio_id,
+                                         'Downloading info json %s' % audio_info_file,
+                                         'Unable to download info file', fatal=True)
+
+        formats = []
+        for bps, k in (('24k', 'play_path_32'), ('64k', 'play_path_64')):
+            if audio_info.get(k):
+                formats.append({
+                    'format_id': bps,
+                    'url': audio_info[k],
+                    'ext': 'm4a',
+                })
+
+        # cover pics kyes like: cover_url', 'cover_url_142'
+        thumbnails = [{'name': k, 'url': audio_info.get(k)} for k in audio_info.keys() if k.startswith('cover_url')]
+
+        audio_uploader_id = audio_info.get('uid')
+
+        if is_m:
+            intro = re.search(r'(?s)<section class=["\']content[^>]+>(.+)</section>', webpage)
```

Comment:
Using spaces in regular expressions is not a good idea. Check [coding conventions](https://github.com/rg3/youtube-dl/blob/master/README.md#youtube-dl-coding-conventions)